learn/getting-started/building-a-data-service/building-a-data-service.md

```diff
@@ -0,0 +1,323 @@
+    }
+}
+
+isolated function removeEmployee(int id) returns int|error {
+    sql:ExecutionResult result = check dbClient->execute(`
+        DELETE FROM Employees WHERE employee_id = ${id}
+    `);
+    return result.affectedRowCount;
+}
+```
+
+## Exposing the Database via an HTTP RESTful API
+
+After you have defined the methods necessary to manipulate the database, you can expose these selectively via an HTTP
+RESTful API. For this, you would first need to import the Ballerina [`HTTP` module](https://central.ballerina.io/ballerina/http).
+
+```ballerina
+import ballerina/http;
+```
+
+### Creating a Service
+Afterwards, you can create a service as follows. This would create an endpoint `/employees` on port `8080` which can 
+be accessed on a browser by visiting `locahost:8080/employees` after executing the command `bal run`.
+
+```ballerina
+service /employees on new http:Listener(8080) {
+}
+```
+
+### Creating Resource Functions
+
+Within this service, you can define resource functions to provide access to the database. The code snippet below 
+demonstrates a resource function that can be used to create a new employee via a `POST` request.
+
+```ballerina
+service /employees on new http:Listener(8080) {
+
+    	isolated resource function post .(@http:Payload Employee emp) returns int|error? {
+            return addEmployee(emp);
+        }
+        
+}
+```
+
+This resource function may be invoked by sending a `POST` request to `http://localhost:8080/employees` with the 
+required data as a JSON payload.
+
+```shell
+curl --location --request POST 'http://localhost:8080/employees/' \
+    --header 'Content-Type: text/plain' \
+    --data-raw '{
+        "employee_id": 6,
+        "first_name": "test",
+        "last_name": "test",
+        "email": "test@test.com",
+        "phone": "882 771 110",
+        "hire_date": {
+            "year": 2021,
+            "month": 12,
+            "day": 16
+        },
+        "manager_id": 1,
+        "job_title": "Sales Manager"
+    }'
+```
+
+
+Similarly, you can define resource functions within the service for each of our use cases as demonstrated below.
+
+```ballerina
+service /employees on new http:Listener(8080) {
+
+    isolated resource function post .(@http:Payload Employee emp) returns int|error? {
+        return addEmployee(emp);
+    }
+    
+    isolated resource function get [int id]() returns Employee|error? {
+        return getEmployee(id);
+    }
+    
+    isolated resource function get .() returns Employee[]|error? {
+        return getAllEmployees();
+    }
+    
+    isolated resource function put .(@http:Payload Employee emp) returns int|error? {
+        return updateEmployee(emp);
+    }
+    
+    isolated resource function delete [int id]() returns int|error? {
+        return removeEmployee(id);       
+    }
+
+}
+```
+
+>**Note:** 
+>This tutorial does not cover topics such as authentication, rate-limiting, usage monitoring, billing, caching, 
```

Comment:
```suggestion
>This tutorial does not cover topics such as authentication, rate-limiting, usage monitoring, billing, caching, 
```
Do we need to mention this explicitly?

Reply:
@daneshk thoughts?

I thought it would be useful to mention this for potential users to know that Ballerina is capable of providing the above as well.

Reply:
Replaced